client/landing/gutenboarding/components/header/index.tsx

```diff
@@ -28,21 +30,53 @@ const Header: React.FunctionComponent = () => {
 
 	const { domain, siteTitle } = useSelect( ( select ) => select( ONBOARD_STORE ).getState() );
 
+	const [ domainSearch ] = useDebounce( siteTitle, 1000 /*ms*/ );
+
+	const recommendedDomain = useSelect(
+		( select ) => {
+			if ( ! domainSearch || domainSearch.length < 2 ) {
+				return;
+			}
+			return select( 'automattic/domains/suggestions' ).getDomainSuggestions( domainSearch, {
+				// Avoid `only_wordpressdotcom` — it seems to fail to find results sometimes
+				include_wordpressdotcom: false,
+				include_dotblogsubdomain: false,
+				quantity: 1, // this will give the recommended domain only
+				locale: i18nLocale,
+			} );
+		},
+		[ domainSearch ]
+	)?.[ 0 ];
+
+	const getDomainElementContent = () => {
+		// If no site title entered (user skips intent gathering)
+		// and no domain was selected, show "Pick a domain"
+		if ( ! siteTitle && ! domain ) {
+			return null;
+		}
+
+		if ( recommendedDomain && ! domain ) {
+			/* translators: domain name is available, eg: "yourname.com is available" */
+			return sprintf( __( '%s is available' ), recommendedDomain.domain_name );
+		}
+
+		return __( 'Choose a domain' );
+	};
+
 	/* eslint-disable wpcalypso/jsx-classname-namespace */
-	const domainElement = domain ? (
-		domain.domain_name
-	) : (
-		<span className="gutenboarding__header-domain-picker-button-domain">
-			{ __( 'Choose a domain' ) }
-		</span>
-	);
+	const domainElement = domain
+		? domain.domain_name
+		: getDomainElementContent() && (
+				<span className="gutenboarding__header-domain-picker-button-domain">
+					{ getDomainElementContent() }
+				</span>
+		  );
 
 	const hideFullHeader = [
 		'IntentGathering',
 		'Domains',
 		'DomainsModal',
 		'Plans',
```

Comment:
`Plans` needs to be removed as well from this exclusion list since it's the actual step in the linear flow.

Reply:
What does this part do / what's the context of comment? :-)

Reply:
It's hiding the domain nudge and plan button from Gutenboarding header on a list of routes. By removing only `PlansModal` from the list, you'll see the domain nudge when clicking Plan button but you won't see it when you reach `Plans` step in the flow.

Reply:
Gotcha, yes let's remove for Plans step as well. 👍

Reply:
done in https://github.com/Automattic/wp-calypso/pull/44655/commits/0e33fc5fff296e2cec07731a8255004ddbc2d448